src/Features/Core/Portable/AddPackage/ParentInstallPackageCodeAction.cs

```diff
@@ -0,0 +1,129 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis.AddImport;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.Packaging;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.AddPackage;
+
+/// <summary>
+/// Data used to create the 'Install Nuget Package' top level code-action. It will have children to 'Install Latest',
+/// 'Install Version 'X' ..., and 'Install with package manager'.
+/// </summary>
+/// <param name="packageSource">The nuget source to use.  Currently this is only <see
+/// cref="PackageSourceHelper.NugetOrgSourceName"/> ("::nuget::")</param>
+/// <param name="packageName">The name of the package to install.</param>
+/// <param name="packageVersionOpt">A preferred version if known.</param>
+/// <param name="textChanges">Additional text changes to make to the <see cref="Document"/>.  Generally, this would be
+/// the import to add if not present.</param>
+internal readonly struct InstallPackageData(string packageSource, string packageName, string packageVersionOpt, ImmutableArray<TextChange> textChanges)
+{
+    public readonly string PackageSource = packageSource;
+    public readonly string PackageName = packageName;
+    public readonly string PackageVersionOpt = packageVersionOpt;
+
+    public readonly ImmutableArray<TextChange> TextChanges = textChanges;
+}
+
+/// <summary>
+/// This is the top level 'Install Nuget Package' code action we show in 
+/// the lightbulb.  It will have children to 'Install Latest', 
+/// 'Install Version 'X' ..., and 'Install with package manager'.
+/// </summary>
+internal sealed class ParentInstallPackageCodeAction : CodeAction.CodeActionWithNestedActions
```

Comment:
this is a move (from nested type to top level type, and to new folder).  Github didn't seem to pick this up.